src/overlays/actors/ovl_Bg_Open_Spot/z_bg_open_spot.c

```diff
@@ -27,25 +28,50 @@ const ActorInit Bg_Open_Spot_InitVars = {
     (ActorFunc)NULL,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80ACB3B0[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 560, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 800, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+void BgOpenSpot_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgOpenSpot* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->texScroll = Lib_SegmentedToVirtual(gSpotlightTexAnim);
+}
+
+void BgOpenSpot_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-extern InitChainEntry D_80ACB3B0[];
+void BgOpenSpot_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgOpenSpot* this = THIS;
+    u32 action;
 
-extern UNK_TYPE D_06001A60;
-extern UNK_TYPE D_06002CE0;
+    if (Cutscene_CheckActorAction(globalCtx, 0x7D)) {
+        action = globalCtx->csCtx.actorActions[Cutscene_GetActorActionIndex(globalCtx, 0x7D)]->action;
+        if (action == 1) {
+            this->actor.draw = NULL;
+            return;
+        } else if (action == 2) {
+            this->actor.draw = BgOpenSpot_Draw;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Open_Spot/BgOpenSpot_Init.s")
+void BgOpenSpot_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    u32 pad;
```

Comment:
```suggestion
    s32 pad;
```
Minor nitpick, but keeping pads all the same type will make them easier to find if/when we need to.